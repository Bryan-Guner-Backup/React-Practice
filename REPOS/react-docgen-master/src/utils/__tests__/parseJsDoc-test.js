/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import parseJsDoc from '../parseJsDoc';

describe('parseJsDoc', () => {
  describe('description', () => {
    it('extracts the method description in jsdoc', () => {
      const docblock = `
        Don't use this!
      `;
      expect(parseJsDoc(docblock)).toMatchSnapshot();
    });
  });
  describe('@param', () => {
    const docBlocks = {
      'extracts jsdoc description': '@param bar test',
      'extracts jsdoc empty description': '@param {string} bar',
      'extracts jsdoc union type param': '@param {string|Object|some[]} bar',
      'extracts jsdoc optional': '@param {string=} bar',
      'extracts jsdoc typed array': '@param {[string, number]} bar',
    };

    Object.keys(docBlocks).forEach(name => {
      const docBlock = docBlocks[name];
      it(name, () => {
        expect(parseJsDoc(docBlock)).toMatchSnapshot();
      });
    });
  });
  describe('@returns', () => {
    const docBlocks = {
      'extracts jsdoc types': '@returns {string}',
      'extracts jsdoc mixed types': '@returns {*}',
      'extracts description from jsdoc': '@returns The number',
      'works with @return': '@return The number',
      'extracts jsdoc typed array': '@param {[string, number]} bar',
    };

    Object.keys(docBlocks).forEach(name => {
      const docBlock = docBlocks[name];
      it(name, () => {
        expect(parseJsDoc(docBlock)).toMatchSnapshot();
      });
    });
  });
});
