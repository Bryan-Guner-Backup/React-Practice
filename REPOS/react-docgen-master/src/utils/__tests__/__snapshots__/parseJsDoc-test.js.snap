// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseJsDoc @param extracts jsdoc description 1`] = `
Object {
  "description": null,
  "params": Array [
    Object {
      "description": "test",
      "name": "bar",
      "optional": false,
      "type": null,
    },
  ],
  "returns": null,
}
`;

exports[`parseJsDoc @param extracts jsdoc empty description 1`] = `
Object {
  "description": null,
  "params": Array [
    Object {
      "description": null,
      "name": "bar",
      "optional": false,
      "type": Object {
        "name": "string",
      },
    },
  ],
  "returns": null,
}
`;

exports[`parseJsDoc @param extracts jsdoc optional 1`] = `
Object {
  "description": null,
  "params": Array [
    Object {
      "description": null,
      "name": "bar",
      "optional": true,
      "type": Object {
        "name": "string",
      },
    },
  ],
  "returns": null,
}
`;

exports[`parseJsDoc @param extracts jsdoc typed array 1`] = `
Object {
  "description": null,
  "params": Array [
    Object {
      "description": null,
      "name": "bar",
      "optional": false,
      "type": Object {
        "elements": Array [
          Object {
            "name": "string",
          },
          Object {
            "name": "number",
          },
        ],
        "name": "tuple",
      },
    },
  ],
  "returns": null,
}
`;

exports[`parseJsDoc @param extracts jsdoc union type param 1`] = `
Object {
  "description": null,
  "params": Array [
    Object {
      "description": null,
      "name": "bar",
      "optional": false,
      "type": Object {
        "elements": Array [
          Object {
            "name": "string",
          },
          Object {
            "name": "Object",
          },
          Object {
            "elements": Array [
              Object {
                "name": "some",
              },
            ],
            "name": "Array",
          },
        ],
        "name": "union",
      },
    },
  ],
  "returns": null,
}
`;

exports[`parseJsDoc @returns extracts description from jsdoc 1`] = `
Object {
  "description": null,
  "params": Array [],
  "returns": Object {
    "description": "The number",
    "type": null,
  },
}
`;

exports[`parseJsDoc @returns extracts jsdoc mixed types 1`] = `
Object {
  "description": null,
  "params": Array [],
  "returns": Object {
    "description": null,
    "type": Object {
      "name": "mixed",
    },
  },
}
`;

exports[`parseJsDoc @returns extracts jsdoc typed array 1`] = `
Object {
  "description": null,
  "params": Array [
    Object {
      "description": null,
      "name": "bar",
      "optional": false,
      "type": Object {
        "elements": Array [
          Object {
            "name": "string",
          },
          Object {
            "name": "number",
          },
        ],
        "name": "tuple",
      },
    },
  ],
  "returns": null,
}
`;

exports[`parseJsDoc @returns extracts jsdoc types 1`] = `
Object {
  "description": null,
  "params": Array [],
  "returns": Object {
    "description": null,
    "type": Object {
      "name": "string",
    },
  },
}
`;

exports[`parseJsDoc @returns works with @return 1`] = `
Object {
  "description": null,
  "params": Array [],
  "returns": Object {
    "description": "The number",
    "type": null,
  },
}
`;

exports[`parseJsDoc description extracts the method description in jsdoc 1`] = `
Object {
  "description": "Don't use this!",
  "params": Array [],
  "returns": null,
}
`;
